File: sub-model/src/main/java/org/github/base/entity/CategoryEntity.java
package org.github.base.entity;

import org.github.base.Model;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableField;
import java.io.Serializable;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * <p>
 * 商品分类 
 * </p>
 *
 * @author JYD_XL
 * @since 2021-04-07
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("category")
@ApiModel(value="CategoryEntity对象", description="商品分类 ")
public class CategoryEntity extends Model<CategoryEntity> {

  /** UID */
  private static final long serialVersionUID = 1L;

  /** 主键 */
  @ApiModelProperty(value = "主键")
  @TableId(value = "id", type = IdType.AUTO)
  private Integer id;

  /** 分类名称 */
  @ApiModelProperty(value = "分类名称")
  @TableField("name")
  private String name;

  /** 分类类型 */
  @ApiModelProperty(value = "分类类型")
  @TableField("type")
  private Integer type;

  /** 父id */
  @ApiModelProperty(value = "父id")
  @TableField("father_id")
  private Integer fatherId;

  /** 图标 */
  @ApiModelProperty(value = "图标")
  @TableField("logo")
  private String logo;

  /** 口号 */
  @ApiModelProperty(value = "口号")
  @TableField("slogan")
  private String slogan;

  /** 分类图 */
  @ApiModelProperty(value = "分类图")
  @TableField("cat_image")
  private String catImage;

  /** 背景颜色 */
  @ApiModelProperty(value = "背景颜色")
  @TableField("bg_color")
  private String bgColor;

  @Override
  protected Serializable pkVal() {
    return this.id;
  }

}

File: sub-mysql-blog-entity/src/main/java/org/github/mysql/blog/base/entity/CategoryEntity.kt
package org.github.mysql.blog.base.entity

import com.baomidou.mybatisplus.annotation.IdType
import com.baomidou.mybatisplus.annotation.TableField
import com.baomidou.mybatisplus.annotation.TableId
import com.baomidou.mybatisplus.annotation.TableName
import java.time.LocalDateTime
import org.github.base.Entity
import io.swagger.annotations.ApiModel
import io.swagger.annotations.ApiModelProperty

/**
 * 
 *
 * @author JYD_XL
 */
@TableName("tb_category")
@ApiModel(value = "CategoryEntity对象", description = "")
open class CategoryEntity : Entity() {

    @TableId(value = "id", type = IdType.AUTO)
    open var id: Int? = null

    /** 分类名 */
    @ApiModelProperty(value = "分类名")
    @TableField("category_name")
    open var categoryName: String? = null

    /** 创建时间 */
    @ApiModelProperty(value = "创建时间")
    @TableField("create_time")
    open var createTime: LocalDateTime? = null

    companion object {

        private const val serialVersionUID = 1L

        const val ID : String = "id"

        const val CATEGORY_NAME : String = "category_name"

        const val CREATE_TIME : String = "create_time"

    }

    override fun pkVal(): java.io.Serializable? {
        return id
    }

    override fun toString(): String {
        return "CategoryEntity{" +
        "id=" + id +
        ", categoryName=" + categoryName +
        ", createTime=" + createTime +
        "}"
    }

}
